src/components/PostsList/PostsList.tsx

```diff
@@ -1,53 +1,83 @@
-import React from 'react';
-
+import React, { useState, useEffect } from 'react';
 import './PostsList.scss';
+import { getAllPosts, getPosts } from '../../api/posts';
+import { Post } from '../../react-app-env';
 
 type Props = {
-  posts: Post[],
-  postId: number,
-  setPostId: (id: number) => void,
+  userSelectedId: string;
+  selectPost: (arg?: Post) => void;
+  post: Post | undefined;
 };
 
 export const PostsList: React.FC<Props> = ({
-  posts,
-  postId,
-  setPostId,
-}) => (
-  <div className="PostsList">
-    <h2>Posts:</h2>
-
-    {posts.length === 0
-      ? <p>Posts not found</p>
-      : (
-        <ul className="PostsList__list" data-cy="postsList">
-          {posts.map(post => (
-            <li className="PostsList__item" key={post.id}>
+  userSelectedId,
+  selectPost,
+  post,
+}) => {
+  const [currentPostList, setPostList] = useState<Post[]>([]);
+
+  const allPosts = async () => {
+    const result = await getAllPosts();
+
+    setPostList(result);
+  };
+
+  const findposts = async () => {
+    const result = await getPosts(userSelectedId);
+
+    setPostList(result);
+  };
+
+  useEffect(() => {
+    findposts();
+  }, [userSelectedId]);
+
+  useEffect(() => {
+    allPosts();
+  }, []);
+
+  return (
+    <div className="PostsList">
+      <>
+        <h2>Posts:</h2>
+        <ul
+          className="PostsList__list"
+          data-cy="postDetails"
+        >
+          {currentPostList.map(po => (
```

Comment:
```suggestion
          {currentPostList.map(currentPost => (
```
`po` is not readable